### ISSUE TITLE
Bar component degrades performance when added to Highlight and Zoom example

### ISSUE THREAD

**Issue by lgreitis** [recharts/recharts#3401]
- [ ] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/highlight-zomm-line-chart-forked-f5q4ct?file=/src/App.tsx)

### Steps to reproduce
Use the Highlight feature how it's intended and you will see bad performance, then comment out Box component (line 198), and you will see better performance.

### What is expected?
The performance shouldn't change that much when rendering Box component or not.

### What is actually happening?
The performance heavily suffers when using Highlight function alongside Box component.

| Environment | Info |
|---|---|
| Recharts | v2.4.3 |
| React | 18.2.0 |
| System | MacOS Ventura 13.2.1 |
| Browser | Firefox 110.0 and Chrome 110.0.5481.177 |

---
When rendering Box chart in another Responsive container, but using the same data the performance also decreases.

**Comment by ckifer** [comment]
Looks like there is a lot of data being rendered here - recharts has some known issues when rendering large datasets.

This also makes sense to me - there are more graphical items there are to render the more work has to be done and the worse the chart will perform. Recharts definitely needs some performance enhancements, especially for large datasets, but this doesn't seem our of the ordinary.

**Comment by lgreitis** [comment]
Makes sense. Surprised that Bar component can drop performance that much, since Line component can easily handle more than 700 items even when there's two of them.

Hope that the Bar component will receive some optimizations someday.

For now a good alternative is Area component with step type, it fits my use case and has good performance.

**Comment by ckifer** [comment]
What are you using to measure performance?

**Comment by ckifer** [comment]
If you want to profile and document what actually ends up taking the longest for you in this case - it would be huge head start to anyone looking to optimize Bar anytime soon

**Comment by lgreitis** [comment]
I'm using React browser extension's profiler, and checking the flamegraph. The Area with Line component rerenders in about 8ms while Bar with Line component takes around 30ms. 

I might dig a little deeper later, but for now I have some deadlines to catch up to.
